operator-wg/whitepaper-community-review/README.md

```diff
@@ -0,0 +1,948 @@
+used to manage off-cluster and cloud resources, cloud IAM integration
+permissions should be configured to prevent cloud account takeover
+from a compromised operator.
+
+_One thing to note_: A “land grab” of privileges - e.g requesting
+significant/administrative access - is not always malicious in
+intent. The developer might not know better or have had the time
+to tune the required permissions to the concept of least privilege.
+Even in the most innocent case, though, it is still a red flag:
+Perhaps the operator is not yet widely used enough for others to
+have encountered the overuse of privileges, and perhaps it is a
+sign of other security weaknesses within the operator. It is advisable
+to proceed with caution if such a “land grab” is found.
+
+**Software provenance**: The “software supply chain” is starting to get
+more attention at time of writing this whitepaper. Consider the
+source for an operator, how it is being installed, and how or why
+a malicious user may want access to a kubernetes cluster. Spend a few minutes
+reviewing an installation script before running it. While the kubectl
+command supports the ability to apply a yaml script directly from
+the public Internet (e.g `kubectl create -f
+https://publicwebsite.com/install/operator.yaml`) it is strongly
+recommended that one first downloads that file locally, review it,
+and then run `kubectl create -f operator.yaml`.
+
+To review the script ask the following questions:
+
+* What is the purpose of this script?
+* What resources are being created by the script? Is this script creating Roles and RoleBindings?
+* What 3rd party sources will the script attempt to use? (e.g.
+container images, other yaml files) How popular and well-maintained
+are the git and docker image repositories? These might be signs of
+a new project, abandoned software which is no longer receiving
+security updates, or indicators of an unofficial repository with
+malicious intent.
+* What privileges does the script attempt to gain? Does the script
+attempt to run container securityContexts with host sharing or
+“privileged mode”?
+
+**Advanced security controls**, such as SELinux, AppArmor, or seccomp
+may be mandated by cluster policy. Open source operators are unlikely
+to have configurations for these Linux security modules, but if
+an organization is familiar with one of these control systems,
+writing the appropriate security configuration for the operator
+should not require significant overhead.
+
+**Operator configuration**: Ideally a project will be “secure by
+default” to increase the likelihood of a secure operator or application
+deployment. Insecure defaults require manual configuration to secure
+the environment. While it may seem like unnecessary work to learn
+configuration  parameters of a new operator, it is usually preferable
+manually adjusting the configuration and/or source code of an
+operator itself to reach the needed level of security.
```

Comment:
```suggestion
**Operator configuration**: Ideally a project will be “secure by default” to increase the likelihood of a secure operator or application deployment. Insecure defaults require manual configuration to secure the environment. While it may seem like unnecessary work to learn the configuration parameters of a new operator, it is usually preferable to manually adjusting the configuration and/or source code of an operator itself to reach the needed level of security.
```